docs(payments): import transaction overview content (#173)
BitcoinDesign/Guide: guide/payments/transactions.md

#### 1. Get recipient address
You need a valid address to send bitcoin. This can be shared by the recipient as a QR code, in plain text or as a payment link.

#### 2. Creation
The wallet application guides the user through collecting the required information (address and amount), and any optional information (what coins to send, fee options).

#### 3. Signing
The transaction needs to be signed by the [private key(s)]({{ '/guide/glossary/#private-key' | relative_url }}) of the input address(es) to be valid. The signing is often done in the same application and at the same time as the creation, but this does not have to be the case.

#### 4. Broadcasting
The transaction is broadcasted to a Bitcoin node, normally the one the wallet is connected to.

#### 5. Validation
The receiving node checks that the transaction is valid. In practice this means confirming that it was signed by the private key of the relevant address.

#### 6. Propagation
Once validated, the node passes the transaction on to other nodes in the network - returning a success message to the originator. The transaction is now in the *memory pool*, and remains there until confirmed.

#### 7. Confirmations

> At some point the transaction is included in a block that is added to the blockchain. This counts as a first confirmation. The recipient can be confident that they will receive the bitcoin, and may even receive a pre-emptive notification from their wallet. How long the first confirmation takes is dependent on many factors, principally - how many other transactions are waiting to be confirmed and how big a fee the transaction will pay. A new block is confirmed roughly every ten minutes.

Reviewer: > At some point the transaction is included in a block that is added to the blockchain.

Perhaps we want to be a bit more specific here on how this happens? Or link to this https://bitcoin.design/guide/getting-started/technology-primer/#how-are-transactions-confirmed?

Author: Ok, I guess the "how it works" can be linked to the getting started, then this block can just focus on how the amount of confirmations impact payment guarantees. 

- 0 confirmations
- first confirmation
- 6th confirmation, as the standard for the transaction to be finalised (is this still the case?)
- more blocks are generated

Each additional block that is accepted by the network counts as an extra confirmation, making the transaction more secure by making it harder to undo. When a transaction has 6 confirmations, it is considered final.

---

## Transaction structure

> The most common Bitcoin transactions contain instructions to move funds from one address to another. Other types of transactions include creating multi-sig wallets or recording data on the blockchain.

Reviewer: Do we expect a person to know what a multi-sig is at this point, or should we refer them somewhere?

Author: Ye good catch on this. Setting up multisig happens off-chain there's no transaction besides the funding which is just as any normal payment to a bitcoin script.

Once created, a transaction needs to be signed with the private key(s) of the sender(s), submitted to a node on the network for validation and propagation to other nodes. It then waits to be verified and included in a newly mined block, after which point it will be recorded on the blockchain. For this service, the sender will pay a fee for every transaction which goes to the node who mined the block as a reward.

Technically, the transaction is data that contains the necessary information to execute on the blockcain. The principal properties are:

> - **Amount** -- the amount of bitcoin, in satoshis

Reviewer: > in satoshis

Do we expect a person to know what satoshi is, maybe reference it here, and is this a consensus to display amounts in `SATS` through the guide?

Reviewer 2: +1 for having SATS as the standard way to describe things. Could just link the first mention of satoshi's to the glossary entry for this.

- **Inputs** -- the source(s)
- **Outputs** -- the destination(s)
- **Version** -- the code version this transaction follows
- **Lock time** -- time until the transaction should be executed, optional

### Input and outputs

> 

Reviewer: Should we have an illustration here? Usually, Inputs and Outputs are better understood visually, in my experience at least.

---
layout: guide
title: Transaction overview
description: An overview of Bitcoin payments, common concepts and the transaction lifecycle.
nav_order: 1
parent: Payments
permalink: /guide/payments/transactions/
main_classes: -no-top-padding
image: /assets/images/guide/payments/overview/page-overview.svg
---

{% include picture.html
   image = "/assets/images/guide/payments/transactions/transactions-desktop.png"
   retina = "/assets/images/guide/payments/transactions/transactions-desktop@2x.png"
   mobile = "/assets/images/guide/payments/transactions/transactions-mobile.png"
   mobileRetina = "/assets/images/guide/payments/transactions/transactions-mobile@2x.png"
   alt-text = "Process of transaction being mined, and new payments being made, mixed and sent to multiple other wallets"
   caption = '3D payments and transaction lifecycle illustration by [Alexa Aker](https://alexaaker.com){:target="_blank" rel="nofollow"}'
   width = 1600
   height = 800
   layout = "full-width"
%}

# Transactions overview

> This page is meant to equip you with the understanding of how bitcoin transactions get created and processed by the Bitcoin network. Bitcoin is a peer-to-peer push-payment system. This means that you can send, or *push*, bitcoin to any address at your will, at any time, without passing through a trusted third party.

Reviewer: ```suggestion
This page is meant to equip you with understanding how bitcoin transactions get created and processed by the Bitcoin network. Bitcoin is a peer-to-peer push-payment system. This means that you can send, or push, bitcoin to any address at your will, at any time, without passing through a trusted third party.
```

This is radically different from the traditional financial system, where it is often possible for others to *pull* and withdraw money from your account (utility companies, financial institutions, merchants, etc.). When you make a payment, it will pass through systems that might delay, control or block the payment.

## Transaction lifecycle

Let's lay out the entire payment process.

{% include picture.html
   image = "/assets/images/guide/payments/transactions/tx-lifecycle-simplified.svg"
   mobile = "/assets/images/guide/payments/transactions/tx-lifecycle-simplified-mobile.svg"
   alt-text = "Flow chart of the process of a transaction getting created and then finally confirmed"
   width = 1350
   height = 242
   layout = "full-width"
%}

#### 1. Get recipient address

> The sender needs a valid address for the payment to be sent to. This can be shared by the recipient as a QR code, in plain text or as a payment link.

Reviewer: ```suggestion
The sender needs a valid address for the payment to be sent to. This can be shared by the recipient as a QR code, in plain text, or as a payment link.
```

#### 2. Creation

> The wallet application guides the sender through collecting the required information (address and amount), and any optional configurations (which coins to send, fee options) in order to create a transaction.

Reviewer: ```suggestion
The wallet application guides the sender through collecting the required information (address and amount) and any optional configurations (which coins to send, fee options) in order to create a transaction.
```

#### 3. Signing
The transaction needs to be signed by the [private key(s)]({{ '/guide/glossary/#private-key' | relative_url }}) of the input [address(es)]({{ '/guide/glossary/#address' | relative_url }}) to be valid. The signing is often done in the same application after the transaction has created and configured, but this does not have to be the case.

#### 4. Broadcasting
The transaction is broadcasted to a Bitcoin node, normally the one the wallet is connected to.

#### 5. Validation

> The receiving node checks that the transaction is valid. In practice this means confirming that it was signed by the private key of the relevant address.

Reviewer: ```suggestion
The receiving node checks that the transaction is valid. In practice, this means confirming that it was signed by the private key of the relevant address.
```

#### 6. Propagation

> Once validated, the node passes the transaction on to other nodes in the network. The transaction is now in the *memory pool*, and remains there until it is mined. At this stage the transaction has 0 confirmations.

Reviewer: ```suggestion
Once validated, the node passes the transaction on to other nodes in the network. The transaction is now in the *memory pool*, and remains there until it is mined. At this stage, the transaction has 0 confirmations.
```

<!-- indicate that a tx with a fee rate that is lower than the current normal can be stuck and even forgotten by miners if it remains in the mempool for too long -->

#### 7. Confirmations
Given that you know [how transactions are confirmed]({{ '/guide/getting-started/technology-primer/#how-are-transactions-confirmed' | relative_url }}), lets look at how the number of confirmations affects the payment settlement.

> Not every miner creates the new block with the same transactions, so some nodes may have a different version of the blockchain than others for a short time. The Bitcoin protocol's main function is to bring all nodes to the same version of the blockchain. Through a process called chain reorganization, nodes remove their incorrect block and update with the winning block as determined by the majority of other nodes.

Reviewer: ```suggestion
Not every miner creates a new block with the same transactions, so some nodes may have a different version of the blockchain than others for a short time. The Bitcoin protocol's main function is to bring all nodes to the same version of the blockchain. Through a process called chain reorganization, nodes remove their incorrect block and update with the winning block as determined by the majority of other nodes.
```

Author: I wanted to convey that at a given epoch - miners are all working to find the same block #

Reviewer: Ah, okay, yeah in that case - my bad. Leave it as is, makes sense.

> There is a slight risk that a transaction with 1 confirmation may revert to 0 confirmaations when a chain reorganisation occurs. Due to this, some parties may require more confirmations for the transaction before providing the product or service.

Reviewer: Standardization to 🇺🇸 - 🇬🇧 English for consistency.
```suggestion
There is a slight risk that a transaction with 1 confirmation may revert to 0 confirmations when a chain reorganization occurs. Due to this, some parties may require more confirmations for the transaction before providing the product or service.
```

Author: damn ye sorry force of habit and growing up with british spelling, i need to setup my text editor.

It is widely accepted that after 6 confirmations, no other reorganizations would occur, and the payment final.

---

## Transaction structure

> Think of a transaction as a file that contains the authorisations to spend some bitcoin, as well as the payment details of the recipient(s). Once miners get the transaction they check all the details once more before they include it into a block.

Reviewer: ```suggestion
Think of a transaction as a file that contains the authorizations to spend some bitcoin, as well as the payment details of the recipient(s). Once miners get the transaction they check all the details once more before they include it into a block.
```

The principal properties are:

- **Amount** -- the amount of bitcoin, in satoshis
- **Inputs** -- the source of funds
- **Outputs** -- the payment or change destination(s)
- **Lock time** -- the earliest the transaction can be broadcasted (optional)
<!-- Purposfully left out RBF as I don't think this format would guide the reader to understanding how it works. Will add in v2 of the page -->

### Input and outputs

In a Bitcoin wallet, funds are often not held in a single address, but more commonly in one address per transaction where you previously received bitcoin. When you are creating a transaction you need to specify which of your addresses you would like to use to fund it. If you need to spend more than what a single address holds, you can specify several. These are called *inputs* to the transaction.

> Likewise, you need to specify the destination address, or addresses for the transaction. These are called *outputs*. should there be more bitcoin in the inputs than are needed for the payment, a new address will be created in your wallet for the remaining change, often called a *change output*.

Reviewer: ```suggestion
Likewise, you need to specify the destination address or addresses for the transaction. These are called *outputs*. should there be more bitcoin in the inputs than are needed for the payment, a new address will be created in your wallet for the remaining change, often called a *change output*.
```

### Transaction fee

> Every transaction needs to pay a fee to incentivise miners to include it in a block. There is no fixed fee for making a transaction as it depends on the amount of data it includes, the amount of other transactions that are trying to get verified, and how much each submitter is prepared to pay. Miners typically pick the transactions that will earn them the highest reward to include in a block.

Reviewer: ```suggestion
Every transaction needs to pay a fee to incentivize miners to include it in a block. There is no fixed fee for making a transaction as it depends on the amount of data it includes, the amount of other transactions that are trying to get verified, and how much each submitter is prepared to pay. Miners typically pick the transactions that will earn them the highest reward to include in a block.
```

> Blocks are also limited in size and new ones are created every 10 minutes on average. This means that if you want a transaction to be confirmed in the next block you might have to pay a relativly high price.

Reviewer: ```suggestion
Blocks are also limited in size, and new ones are created every 10 minutes on average. This means that if you want a transaction to be confirmed in the next block, you might have to pay a relatively high price.
```
